### PR TITLE
[5.9] Package.swift: make package name consistent with repo name

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Rest of SwiftPM packages provided by Apple follow a dash-case naming scheme consistent with repository name, usually with `swift-` prefix.

This shouldn't have an impact on SwiftSyntax clients. For example `swift-format` already uses `package: "swift-syntax"` argument in its `Package.swift` for specifying the dependency.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by ahoppen — APPROVED** [review]

**Comment by kimdv** [comment]
@swift-ci please test windows

**Comment by MaxDesiatov** [comment]
Would this require more approvals before merging?

**Review by bnbarham — APPROVED** [review]

**Pull request merged**
